core/src/main/java/com/scalar/db/storage/multistorage/MultiStorageAdmin.java

```diff
@@ -200,6 +207,48 @@ public void importTable(String namespace, String table) throws ExecutionExceptio
     getAdmin(namespace, table).importTable(namespace, table);
   }
 
+  @Override
+  public Set<String> getNamespaceNames() throws ExecutionException {
+    // Only return existing namespaces that are listed in the namespace mapping configuration or
+    // when they belong to the default storage
+    //
+    // For example, if the storages contain the following namespaces :
+    // - mysql : mysqlStorageAdmin.getNamespaceNames() = [ns1, ns2]
+    // - cassandra : cassandraStorageAdmin.getNamespaceNames() = [ns3]
+    // - cosmos : cosmosStorageAdmin.getNamespaceNames() = [ns4, ns5]
+    // And the default storage is cosmos :
+    // - scalar.db.multi_storage.default_storage=cosmos
+    // And the namespace mapping set in the configuration is :
+    // - scalar.db.multi_storage.namespace_mapping=ns1:mysql,ns2:cassandra,ns3:cassandra
+    //
+    // Then multiStorageAdmin.getNamespaceNames() = [ns1, ns3, ns4, ns5]
+    // The reasoning is:
+    // - ns1 is present in the mysql storage and listed in the mapping belonging to mysql
+    //     => returned
+    // - ns2 is present in the mysql storage but listed in the mapping belonging to cassandra
```

Comment:
Sorry, I don't fully get this. 
Is this a misconfiguration by users? (ns2 is defined in MySQL but the namespace mapping says it's mapped to Cassandra?)
It would be great if you could elaborate on this.

Reply:
> Is this a misconfiguration by users? (ns2 is defined in MySQL but the namespace mapping says it's mapped to Cassandra?)

It can be a misconfiguration or deliberate to do some testing (even though it's probably not safe to do so). 

This is a corner case that is probably unlikely to happen, but I included it in this explanation to highlight only namespaces strictly defined in the configuration are returned by this method.

Reply:
Let me clarify.
So, there are two apps that share the same ScalarDB Cluster.
One app (app1) defines ns1 and ns2 in its config, and the other (app2) defines ns3 and ns4 in its config.
But, the actual database defines ns1, ns2, ns3, and ns4.
In this case, app1 returns ns1 and ns2, and app2 returns ns3 and ns4 by `getNamespaceNames`?
I'm not sure if it is an expected behavior.

Reply:
> In this case, app1 returns ns1 and ns2, and app2 returns ns3 and ns4 by getNamespaceNames?

Correct. Yes, that is the expected behavior.